node-eclib.js

```diff
@@ -62,30 +62,38 @@ ECLib.prototype = {
      */
     init: function(callback) {
         var instance_descriptor_id = -1;
-        var err = {};
+        var err = null;
         var o = this.opt;
         if (util.validateInstance(this.opt)) {
             instance_descriptor_id = addon.EclCreate(o.bc_id, o.k, o.m, o.w,
                     o.hd, o.ct);
 
             if (instance_descriptor_id <= 0) {
-                err.errorcode = instance_descriptor_id;
-                err.message = util.getErrorMessage(instance_descriptor_id);
+                err = {
+                    errorcode: instance_descriptor_id,
+                    message: util.getErrorMessage(instance_descriptor_id),
+                };
             } else {
                 this.ins_id = instance_descriptor_id;
             }
 
         } else {
-            err.errorcode = enums.ErrorCode.EINVALIDPARAMS;
-            err.message = util.getErrorMessage(err.errorcode);
+            err = {
+                errorcode: enums.ErrorCode.EINVALIDPARAMS,
+                message: util.getErrorMessage(enums.ErrorCode.EINVALIDPARAMS),
+            };
             instance_descriptor_id = err.errorcode;
         }
 
         if (!callback) {
+            if (instance_descriptor_id < 0) {
+                throw err.message;
```

Comment:
it shows error message that is useful for user, instead of all error code and message